docs/unidling.md

```diff
@@ -0,0 +1,183 @@
+# Unidling Pods
+## Abstract
+
+Unidling is an event-driven feature that allows users to automatically create destination pods for
+an idled service when a connection is made to it.
+
+## Motivation
+
+Not all containers in a Kubernetes cluster will be active at all times.  When all of the containers
+resolving to a service are inactive, new ones must be created when a new request for that service
+is received.  The process of creating new pods for an idled service in response to a new request to
+that service is called "unidling".  This proposal covers how unidling should be performed in
+Kubernetes.
+
+We refer to the process of determining which containers to stop and stopping them as "idling"; it
+will be dealt with in a separate proposal and is out of scope for this proposal.  
+
+## Lexicon
+
+- **Destination pods**: the backend pods (or other backends as necessary) that a service points to
+- **Idled service**: a service with zero endpoints
+
+## Constraints and Assumptions
+
+- Unidling should be orthogonal to autoscaling and idling
+- No specific idling mechanism or autoscaler is assumed
+- The case where a service is backed by pods which are not managed by a replication controller: 
+  - Not covered by this proposal at all
+  - Requests to unidle these service should be gracefully ignored by components scoped to this
+    proposal
+  - A separate unidler may be written for these
+
+## Use Cases
+
+The following use cases are explored by this proposal:
+
+1.  As a cluster operator, I want a new requests to an idled service to trigger the creation of new
+    destination pods for that service
```

Comment:
You need to explain *why* something is idled to explain why it would be unidled :)

----- Original Message -----
> > +## Constraints and Assumptions
> > +
> > +- Unidling should be orthogonal to autoscaling and idling
> > +- No specific idling mechanism or autoscaler is assumed
> > +- The case where a service is backed by pods which are not managed by a
> > replication controller:
> > +  - Not covered by this proposal at all
> > +  - Requests to unidle these service should be gracefully ignored by
> > components scoped to this
> > +    proposal
> > +  - A separate unidler may be written for these
> > +
> > +## Use Cases
> > +
> > +The following use cases are explored by this proposal:
> > +
> > +1.  As a cluster operator, I want a new requests to an idled service to
> > trigger the creation of new
> > +    destination pods for that service
> 
> Should that be a use-case for this doc, since we don't handle idling with
> this?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/pmorie/kubernetes/pull/2/files#r22492156